test/evm/hardhat/MerkleLib.Proofs.ts

```diff
@@ -149,4 +151,62 @@ describe("MerkleLib Proofs", async function () {
     expect(() => merkleTree.getHexProof(invalidLeaf)).to.throw();
     expect(await merkleLibTest.verifyV3SlowRelayFulfillment(root, invalidLeaf, proof)).to.equal(false);
   });
+  it("Legacy V3SlowFill produces the same MerkleLeaf", async function () {
+    const chainId = randomBigNumber(2).toNumber();
+    const relayData: V3RelayData = {
+      depositor: randomBytes32(),
+      recipient: randomBytes32(),
+      exclusiveRelayer: randomBytes32(),
+      inputToken: randomBytes32(),
+      outputToken: randomBytes32(),
+      inputAmount: randomBigNumber(),
+      outputAmount: randomBigNumber(),
+      originChainId: randomBigNumber(2).toNumber(),
+      depositId: randomBigNumber(2),
+      fillDeadline: randomBigNumber(2).toNumber(),
+      exclusivityDeadline: randomBigNumber(2).toNumber(),
+      message: ethers.utils.hexlify(ethers.utils.randomBytes(1024)),
+    };
+    const slowLeaf: V3SlowFill = {
+      relayData,
+      chainId,
+      updatedOutputAmount: relayData.outputAmount,
+    };
+    const legacyRelayData: V3RelayData = {
+      ...relayData,
+      depositor: bytes32ToAddress(relayData.depositor),
+      recipient: bytes32ToAddress(relayData.recipient),
+      exclusiveRelayer: bytes32ToAddress(relayData.exclusiveRelayer),
+      inputToken: bytes32ToAddress(relayData.inputToken),
+      outputToken: bytes32ToAddress(relayData.outputToken),
+    };
+    const legacySlowLeaf: V3SlowFill = {
+      relayData: legacyRelayData,
+      chainId: slowLeaf.chainId,
+      updatedOutputAmount: slowLeaf.updatedOutputAmount,
+    };
+
+    const paramType = await getParamType("MerkleLibTest", "verifyV3SlowRelayFulfillment", "slowFill");
+    const hashFn = (input: V3SlowFill) => keccak256(defaultAbiCoder.encode([paramType!], [input]));
+    const merkleTree = new MerkleTree<V3SlowFill>([slowLeaf], hashFn);
+    const root = merkleTree.getHexRoot();
+
+    const legacyHashFn = (input: V3SlowFill) =>
+      keccak256(
+        defaultAbiCoder.encode(
+          [
+            "tuple(" +
+              "tuple(address depositor, address recipient, address exclusiveRelayer, address inputToken, address outputToken, uint256 inputAmount, uint256 outputAmount, uint256 originChainId, uint32 depositId, uint32 fillDeadline, uint32 exclusivityDeadline, bytes message) relayData," +
+              "uint256 chainId," +
+              "uint256 updatedOutputAmount" +
+              ")",
+          ],
+          [input]
+        )
+      );
+    const merkleTreeLegacy = new MerkleTree<V3SlowFill>([legacySlowLeaf], legacyHashFn);
+    const legacyRoot = merkleTreeLegacy.getHexRoot();
+
+    expect(legacyRoot).to.equal(root);
```

Comment:
Any idea why this line would fail? Assuming I have the type described correctly above on line 198, do merkle trees have other reasons why they could be different?